src/search.js

```diff
@@ -51,7 +50,14 @@ var Search = React.createClass({
     const idealSearchBoxWidth = Math.max(17, this.state.terms && this.state.terms.length*1.1 || 0)
     const formProps = universal ? {action: "/search/", method: "get"} : {action: ''}
     const simpleSearchBox = <div className='search-wrapper' style={{width: idealSearchBoxWidth + 'em'}}>
-      <form {...formProps}><input className='search-input' type="search"
+      <form {...formProps} role="search">
+
+        <label htmlFor='search-input' className="visuallyhidden">Search</label>
```

Comment:
Adding a label for this input both using arial-label as well as a label that is visuallyhidden (this uses material-ui's class to marginLeft it off the screen)